frontend/app/components/session-timeout.tsx

```diff
@@ -0,0 +1,67 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { useTranslation } from 'react-i18next';
+import { type IIdleTimerProps, useIdleTimer } from 'react-idle-timer';
+
+import { Button } from './buttons';
+import { Dialog, DialogContent, DialogFooter, DialogHeader, DialogTitle } from '~/components/dialog';
+import { getTypedI18nNamespaces } from '~/utils/locale-utils';
+
+const i18nNamespaces = getTypedI18nNamespaces('gcweb');
+
+export interface SessionTimeoutProps extends Pick<IIdleTimerProps, 'promptBeforeIdle'>, Pick<IIdleTimerProps, 'timeout'> {}
+
+const SessionTimeout = ({ promptBeforeIdle, timeout }: SessionTimeoutProps) => {
+  const { t } = useTranslation(i18nNamespaces);
+  const [modalOpen, setModalOpen] = useState(false);
+  const [timeRemaining, setTimeRemaining] = useState('');
+
+  const handleOnIdle = () => {
+    window.location.href = '/auth/logout';
+  };
+
+  const handleOnIdleContinueSession = () => {
+    setModalOpen(false);
+    reset();
+  };
+
+  const { reset, getRemainingTime } = useIdleTimer({
+    onIdle: handleOnIdle,
+    onPrompt: () => setModalOpen(true),
+    promptBeforeIdle: promptBeforeIdle ?? 5 * 60 * 1000, //5 minutes
+    timeout: timeout ?? 20 * 60 * 1000, //20 minutes
```

Comment:
Both of these should be config variables that are sourced from `getEnv()`. Expose the through a loader.

Reply:
this is another task. will update it in another PR.